### PR TITLE
NO-ISSUE: ci(playwright): add Jaeger trace analysis step

### PR DESCRIPTION
**Pull request by jbpratt**
## Summary

- Add a `bash+jq` script (`.github/scripts/analyze-jaeger-traces.sh`) that analyzes the Jaeger traces already exported during Playwright CI runs
- Surfaces actionable signals as GitHub Actions warnings and step summary tables:
  - **Error spans** — 5xx HTTP responses and unhandled exceptions (4xx excluded as expected behavior)
  - **Operation name cardinality** — warns if >100 unique operation names (route parameter leakage)
  - **Orphaned spans** — broken trace context propagation
  - **Slow DB queries** — PostgreSQL spans exceeding 500ms
- Exports `trace-metrics.json` baseline artifact for future cross-run regression comparison
- Runs with `continue-on-error: true` so it's advisory, not blocking

Validated against real trace data from CI run #25060534039 (1000 traces, 1738 spans): correctly identified 1 error span (503 health check during startup), all structural checks passed.

## Test plan

- [x] Tested locally against real CI trace data (`traces.json` from run 25060534039)
- [x] Verified edge cases: empty data, missing file, malformed JSON
- [x] Verified step summary markdown renders correctly
- [x] Verified `trace-metrics.json` output schema
- [ ] Verify CI step appears in Playwright workflow run with expected output

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/scripts/analyze-jaeger-traces.sh [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  (.spans | map(.spanID)) as $ids |
+  .spans[] |
+  select(
+    any(.references[]?; .refType == "CHILD_OF" and (.spanID as $pid | $ids | any(. == $pid) | not))
+  )
+] | length' "$TRACES_FILE")
+
+# ── Check 4: Slow DB queries (>500ms = >500000µs) ──
+slow_queries_json=$(jq '[.data[] as $trace | $trace.spans[] |
+  select(any(.tags[]; .key == "db.system") and .duration > 500000) |
```
[the remark this answers is not in the document]
Comment (jbpratt) [reply .github/scripts/analyze-jaeger-traces.sh:58]: pg is the one only supported

File: .github/scripts/analyze-jaeger-traces.sh [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+fi
+
+# ── Check 1: Error spans (5xx and exception-only, excluding expected 4xx) ──
+error_spans_json=$(jq '[.data[] as $trace | $trace.spans[] |
+  select(
+    any(.tags[]; .key == "otel.status_code" and .value == "ERROR")
+    or any(.tags[]; .key == "error" and (.value == true))
+  ) |
+  ([.tags[] | select(.key == "http.status_code") | .value | tonumber] | first // 500) as $status |
+  select($status >= 500) |
```
[the remark this answers is not in the document]
Comment (Marcusk19) [reply .github/scripts/analyze-jaeger-traces.sh:33]: @jbpratt is this valid?

[the remark this answers is not in the document]
Comment (jbpratt) [reply .github/scripts/analyze-jaeger-traces.sh:33]: Sorry no, 
```
● Finding 1 is not needed. In real data, every 5xx span already has both otel.status_code=ERROR and error=true tags — this is standard
  FlaskInstrumentor behavior. The current filter correctly catches all 5xx errors. The suggested rewrite would be a no-op at best, and at worst could
   introduce false positives.
   ```

**Review by Marcusk19 — APPROVED** [review]
